CMakeLists.txt

```diff
@@ -44,11 +44,12 @@ add_subdirectory("utility")
 add_subdirectory("isobus")
 add_subdirectory("hardware_integration")
 
-if(NOT NO_EXAMPLES)
-    add_subdirectory("examples/vt_version_3_object_pool")
-    add_subdirectory("examples/transport_layer")
-    add_subdirectory("examples/diagnostic_protocol")
-    add_subdirectory("examples/pgn_requests")
+option(BUILD_EXAMPLES "Set to ON to enable building of examples from top level" OFF)
```

Comment:
This is the right way to do options, it shows up in ccmake and gives the user a hint about what it does. Current behavior is to not build examples. Let me know what you want for default behavior.

Reply:
I did not know about this, nice. I think default should be to not build the examples.